README.md

```diff
@@ -2,51 +2,102 @@
 
 [![npm](http://img.shields.io/npm/v/objectfn.svg?style=flat)](https://badge.fury.io/js/objectfn) [![tests](http://img.shields.io/travis/jescalan/objectfn/master.svg?style=flat)](https://travis-ci.org/jescalan/objectfn) [![dependencies](http://img.shields.io/david/jescalan/objectfn.svg?style=flat)](https://david-dm.org/jescalan/objectfn) [![coverage](http://img.shields.io/coveralls/jescalan/objectfn.svg?style=flat)](https://coveralls.io/github/jescalan/objectfn)
 
-Map, reduce, forEach, and filter for objects. Lazy evaluation, no dependencies.
+`map`, `reduce`, `forEach`, and `filter` for plain objects. Lazy evaluation, supports functional and imperative syntax, no dependencies.
 
 ### Why should you care?
 
 I wanted a library that has no dependencies and gives me the basic map/reduce/filter for use on objects. Any existing library I found has boatloads of dependencies, provides tons more extra tools, and/or is unmaintained. So here's `ObjectFn`, just for you!
 
 Also, big props to [@declandewet](https://github.com/declandewet) for the initial implementation of this library!
 
+### Requirements
+
+- [Node.js v.6+](http://nodejs.org)
+
 ### Installation
 
-`npm install objectfn -S`
+Using a terminal:
 
-> **Note:** This project is compatible with node v6+ only
+```sh
+$ npm install objectfn -S
+```
 
 ### Usage
 
-Very straightforward usage. Just import what you need and use it on an object.
+Usage is straightforward. Just import what you need and use it on an object.
+
+#### Imperative style
+
+Takes data first, callback last.
 
 ```js
 const {map, reduce, filter, forEach} = require('objectfn')
 
 const obj = { foo: 'bar', wow: 'doge' }
 
-map(obj, (val, key) => val.toUpperCase())
+const upcasedValues = map(obj, (val) => val.toUpperCase())
+console.log(upcasedValues)
```

Comment:
I've developed a habit of making code examples runnable on tonicdev.com but i see there are [other ways of doing this](https://tonicdev.uservoice.com/knowledgebase/articles/765846-how-do-i-customize-the-example-for-my-npm-package) so I've removed the boilerplate. I also spotted some errors in the example which I've fixed